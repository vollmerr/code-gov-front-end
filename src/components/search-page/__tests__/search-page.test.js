import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import SearchPage from 'components/search-page';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Testing <SearchPage /> component', () => {
  const props = {
    onFilterBoxChange: jest.fn(),
    onFilterTagClick: jest.fn(),
    onSortChange: jest.fn(),
    saveFilterData: jest.fn(),
    updatePage:  jest.fn(),
    searchParams: {
      filters: [],
      page: 1,
      size: 10,
      sort: "data_quality"
    },
    searchResults: {
      params: {
        page: 1,
        size: 10,
        sort: "data_quality",
        agencies: [],
        languages: [],
        licenses: [],
        usageTypes: []
      },
      repos: [{
        agency: {name: "Test Agency 1", acronym: "TA1", website: "https://TA1.ca.gov/", codeUrl: "https://raw.githubusercontent.com/cdt-bpa/code.ca.gov-remote-files/master/CWDS.json", fallback_file: "CWDS.json"},
        contact: {name: "", email: ""},
        date: {created: "2017-07-28T07:00:00.000Z", lastModified: "2018-07-25T07:00:00.000Z", metadataLastUpdated: "2019-01-14T08:00:00.000Z"},
        description: "Jenkins acceptance tests repo",
        homepageURL: "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
        laborHours: 0,
        languages: [],
        name: "jenkins-acceptance-test-builds",
        organization: "CWDS",
        permissions: {licenses: Array(1), usageType: "openSource"},
        repoID: "cwds_cwds_jenkins_acceptance_test_builds1",
        repositoryURL: "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
        score: 8.299999999999999,
        searchScore: 14.710426,
        tags: [""],
        vcs: "git"
      },
      {
        agency: {name: "Test Agency 2", acronym: "TA2", website: "https://TA2.ca.gov/", codeUrl: "https://raw.githubusercontent.com/cdt-bpa/code.ca.gov-remote-files/master/CWDS.json", fallback_file: "CWDS.json"},
        contact: {name: "", email: ""},
        date: {created: "2017-07-28T07:00:00.000Z", lastModified: "2018-07-25T07:00:00.000Z", metadataLastUpdated: "2019-01-14T08:00:00.000Z"},
        description: "Jenkins acceptance tests repo",
        homepageURL: "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
        laborHours: 0,
        languages: [],
        name: "jenkins-acceptance-test-builds",
        organization: "CWDS",
        permissions: {licenses: Array(1), usageType: "openSource"},
        repoID: "cwds_cwds_jenkins_acceptance_test_builds2",
        repositoryURL: "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
        score: 8.299999999999999,
        searchScore: 14.710426,
        tags: [""],
        vcs: "git"
      }
    ],
      total: 1
    },
    filters: {
      agencies: [{name: 'test-name', value: 'TV'}],
      languages: [{name: 'test-name', value: 'TV'}],
      licenses: [{name: 'test-name', value: 'TV'}],
      usageTypes: [{name: 'test-name', value: 'TV'}]
    }
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchPage store={store} />);
  });
  it('renders SearchPage', () => {
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders SearchPage', () => {
    wrapper = mount(<Provider store={store}><Router><SearchPage /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})