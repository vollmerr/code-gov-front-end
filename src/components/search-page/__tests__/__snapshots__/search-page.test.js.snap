// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchPage /> component should render correctly 1`] = `
<SearchPage
  boxes={
    Object {
      "agencies": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "languages": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "licenses": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "usageTypes": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
    }
  }
  filterTags={Array []}
  filteredResults={Array []}
  filters={
    Object {
      "agencies": Array [
        Object {
          "name": "test-name",
          "value": "TV",
        },
      ],
      "languages": Array [
        Object {
          "name": "test-name",
          "value": "TV",
        },
      ],
      "licenses": Array [
        Object {
          "name": "test-name",
          "value": "TV",
        },
      ],
      "usageTypes": Array [
        Object {
          "name": "test-name",
          "value": "TV",
        },
      ],
    }
  }
  onFilterBoxChange={[Function]}
  onFilterTagClick={[Function]}
  onSortChange={[Function]}
  saveFilterData={[Function]}
  searchParams={
    Object {
      "filters": Array [],
      "page": 1,
      "size": 10,
      "sort": "data_quality",
    }
  }
  searchResults={
    Object {
      "params": Object {
        "agencies": Array [],
        "languages": Array [],
        "licenses": Array [],
        "page": 1,
        "size": 10,
        "sort": "data_quality",
        "usageTypes": Array [],
      },
      "repos": Array [
        Object {
          "agency": Object {
            "acronym": "TA1",
            "codeUrl": "https://raw.githubusercontent.com/cdt-bpa/code.ca.gov-remote-files/master/CWDS.json",
            "fallback_file": "CWDS.json",
            "name": "Test Agency 1",
            "website": "https://TA1.ca.gov/",
          },
          "contact": Object {
            "email": "",
            "name": "",
          },
          "date": Object {
            "created": "2017-07-28T07:00:00.000Z",
            "lastModified": "2018-07-25T07:00:00.000Z",
            "metadataLastUpdated": "2019-01-14T08:00:00.000Z",
          },
          "description": "Jenkins acceptance tests repo",
          "homepageURL": "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
          "laborHours": 0,
          "languages": Array [],
          "name": "jenkins-acceptance-test-builds",
          "organization": "CWDS",
          "permissions": Object {
            "licenses": Array [
              undefined,
            ],
            "usageType": "openSource",
          },
          "repoID": "cwds_cwds_jenkins_acceptance_test_builds1",
          "repositoryURL": "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
          "score": 8.299999999999999,
          "searchScore": 14.710426,
          "tags": Array [
            "",
          ],
          "vcs": "git",
        },
        Object {
          "agency": Object {
            "acronym": "TA2",
            "codeUrl": "https://raw.githubusercontent.com/cdt-bpa/code.ca.gov-remote-files/master/CWDS.json",
            "fallback_file": "CWDS.json",
            "name": "Test Agency 2",
            "website": "https://TA2.ca.gov/",
          },
          "contact": Object {
            "email": "",
            "name": "",
          },
          "date": Object {
            "created": "2017-07-28T07:00:00.000Z",
            "lastModified": "2018-07-25T07:00:00.000Z",
            "metadataLastUpdated": "2019-01-14T08:00:00.000Z",
          },
          "description": "Jenkins acceptance tests repo",
          "homepageURL": "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
          "laborHours": 0,
          "languages": Array [],
          "name": "jenkins-acceptance-test-builds",
          "organization": "CWDS",
          "permissions": Object {
            "licenses": Array [
              undefined,
            ],
            "usageType": "openSource",
          },
          "repoID": "cwds_cwds_jenkins_acceptance_test_builds2",
          "repositoryURL": "https://github.com/ca-cwds/jenkins-acceptance-test-builds",
          "score": 8.299999999999999,
          "searchScore": 14.710426,
          "tags": Array [
            "",
          ],
          "vcs": "git",
        },
      ],
      "total": 1,
    }
  }
  selectedPage={1}
  sortOptions={
    Array [
      Object {
        "label": "Data Quality",
        "selected": true,
        "value": "data_quality",
      },
      Object {
        "label": "A-Z",
        "selected": false,
        "value": "a-z",
      },
      Object {
        "label": "Last Updated",
        "selected": false,
        "value": "last_update",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  total={1}
  updatePage={[Function]}
/>
`;
